pkg/controller/actions/deploy/action_deploy.go

```diff
@@ -156,10 +157,19 @@ func (a *Action) deploy(
 		return false, nil
 	}
 
+	fo := a.fieldOwner
+	if fo == "" {
+		fo = rr.OwnerName
+	}
+
 	resources.SetLabels(&obj, a.labels)
 	resources.SetAnnotations(&obj, a.annotations)
 	resources.SetAnnotations(&obj, deployAnnotations)
 
+	if resources.GetLabel(&obj, labels.ComponentPartOf) == "" && fo != "" {
```

Comment:
could there still be a chance that `fo == ""` after line 161 ?

Reply:
It should not, but as today you can do `rr.OwnerName = ""`  as part of any action, so this has been added as an extra check. We probably need to add a getter for `OwnerName` (and probably other  `ReconciliationRequest`  fields) in a subsequent PR